### ISSUE TITLE
invalid stream header: ED000573

### ISSUE THREAD

**Issue by thewolt** [gradle/gradle#4933]
Gradle will fail with the following error if a `null` message is given on a `logger.warn`.

```
Cause: invalid stream header: ED000573
```

Well, this is the only error message written in the intellij console.

### Expected Behavior
This should not crash the gradle daemon

### Current Behavior
The gradle daemon fail with the following stack:
```
Caused by: org.gradle.internal.remote.internal.RecoverableMessageIOException: Could not read message from '/0:0:0:0:0:0:0:1:40543'.
        at org.gradle.internal.remote.internal.inet.SocketConnection.receive(SocketConnection.java:86)
        at org.gradle.launcher.daemon.client.DaemonClientConnection.receive(DaemonClientConnection.java:75)
        ... 31 more
Caused by: java.io.StreamCorruptedException: invalid stream header: ED000573
        at java.io.ObjectInputStream.readStreamHeader(ObjectInputStream.java:862)
        at java.io.ObjectInputStream.<init>(ObjectInputStream.java:354)
        at org.gradle.internal.io.ClassLoaderObjectInputStream.<init>(ClassLoaderObjectInputStream.java:27)
        at org.gradle.internal.serialize.ExceptionReplacingObjectInputStream.<init>(ExceptionReplacingObjectInputStream.java:39)
        at org.gradle.internal.serialize.Message.receive(Message.java:54)
        at org.gradle.internal.serialize.BaseSerializerFactory$ThrowableSerializer.read(BaseSerializerFactory.java:240)
        at org.gradle.internal.serialize.BaseSerializerFactory$ThrowableSerializer.read(BaseSerializerFactory.java:238)
        at org.gradle.internal.logging.serializer.LogEventSerializer.read(LogEventSerializer.java:56)
        at org.gradle.internal.logging.serializer.LogEventSerializer.read(LogEventSerializer.java:26)
        at org.gradle.internal.serialize.DefaultSerializerRegistry$TaggedTypeSerializer.read(DefaultSerializerRegistry.java:144)
        at org.gradle.launcher.daemon.protocol.DaemonMessageSerializer$OutputMessageSerializer.read(DaemonMessageSerializer.java:141)
        at org.gradle.launcher.daemon.protocol.DaemonMessageSerializer$OutputMessageSerializer.read(DaemonMessageSerializer.java:127)
        at org.gradle.internal.serialize.DefaultSerializerRegistry$TaggedTypeSerializer.read(DefaultSerializerRegistry.java:144)
        at org.gradle.internal.serialize.Serializers$StatefulSerializerAdapter$1.read(Serializers.java:36)
        at org.gradle.internal.remote.internal.inet.SocketConnection.receive(SocketConnection.java:79)
        ... 32 more
```

### Context
The issue here is that we do not control what logs some thirdparty library would send. 

In our particular case, it was this code inside the `discoverGitExe` method of `org.eclipse.jgit.util.FS_Win32`.

```
try {
  w = readPipe(userHome(),
      new String[]{"bash", "--login", "-c", "which git"}, // //$NON-NLS-1$ //$NON-NLS-2$ //$NON-NLS-3$ //$NON-NLS-4$
      Charset.defaultCharset().name());
} catch (CommandFailedException e) {
  LOG.warn(e.getMessage());  // <--- the message is null
  return null;
}
```

Note that it was really complicated to spot what was going on as the bug was this issue was only happening with the intellij gradle integration plugin.

### Steps to Reproduce (for bugs)

Create a project with the following `build.gradle`

```
logger.warn(null)
```

### Your Environment

Reproduced with gradle-4.3.1 and gradle-4.6

**Comment by 3ll3d00d** [comment]
encountered this with gradle 4.7, in my case also from git but a misconfiguration this time

```
> git config --list --show-origin 
fatal: unable to access 'H://.gitconfig': Invalid argument  
```

**Comment by cliffred** [comment]
This is fixed in Gradle 5, see #6661.

**Comment by marcphilipp** [comment]
Thanks, @cliffred! 👍
